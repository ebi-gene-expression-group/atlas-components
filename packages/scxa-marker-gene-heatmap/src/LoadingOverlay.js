import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import URI from 'urijs'

import flaskLoaderSvg from './svg/flask-loader.svg'

const TranslucentOverlay = styled.div`
  display: ${props => props.show ? `flex` : `none`};
  position: absolute;
  top: 0; left: 0; right: 0; bottom: 0;
  background: rgba(255,255,255,0.8);
  align-items: center;
  justify-content: center;
  flex-direction: column;
 `

const LoadingOverlay = ({show, resourcesUrl}) =>
  <TranslucentOverlay show={show}>
    <p>Loading, please wait...</p>
    <img src={URI(flaskLoaderSvg, resourcesUrl).toString()}/>
  </TranslucentOverlay>

LoadingOverlay.propTypes = {
  show: PropTypes.bool.isRequired,
  resourcesUrl: PropTypes.string
}

LoadingOverlay.defaultProps = {
  resourcesUrl: ``
}

export default LoadingOverlay
