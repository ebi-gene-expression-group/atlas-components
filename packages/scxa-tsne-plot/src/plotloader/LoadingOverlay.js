import React from 'react'
import PropTypes from 'prop-types'

import URI from 'urijs'

const LoadingOverlay = (props) =>
  <div style={{
    display: props.show ? `flex` : `none`,
    position: `absolute`,
    top: 0, left: 0, right: 0, bottom: 0,
    background: `rgba(255,255,255,0.8)`,
    alignItems: `center`,
    justifyContent: `center`
  }}>
    <div style={{textAlign: `center`}}>
      <p>Loading, please wait...</p>
      <img src={URI(require(`./svg/flask-loader.svg`), props.resourcesUrl).toString()}/>
      <p><small>Powered by <a href={`https://loading.io`}>loading.io</a></small></p>
    </div>
  </div>

LoadingOverlay.propTypes = {
  show: PropTypes.bool.isRequired,
  resourcesUrl: PropTypes.string
}

LoadingOverlay.defaultProps = {
  resourcesUrl: ``
}

export default LoadingOverlay
