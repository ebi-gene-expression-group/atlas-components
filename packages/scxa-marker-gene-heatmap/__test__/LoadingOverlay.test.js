import React from 'react'
import { render } from '@testing-library/react'

import LoadingOverlay from '../src/LoadingOverlay'

describe(`LoadingOverlay`, () => {
  test(`matches snapshot when shown`, () => {
    const { asFragment } = render(<LoadingOverlay show={true} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test(`matches snapshot when hidden`, () => {
    const { asFragment } = render(<LoadingOverlay show={false} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
