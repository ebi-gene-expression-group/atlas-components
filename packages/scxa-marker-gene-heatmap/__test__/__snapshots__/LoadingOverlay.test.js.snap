// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay matches snapshot when hidden 1`] = `
<LoadingOverlay
  resourcesUrl=""
  show={false}
>
  <styled.div
    show={false}
  >
    <div
      className="sc-bdnylx bqQJZW"
    >
      <p>
        Loading, please wait...
      </p>
      <img
        src="test-file-stub"
      />
    </div>
  </styled.div>
</LoadingOverlay>
`;

exports[`LoadingOverlay matches snapshot when shown 1`] = `
<LoadingOverlay
  resourcesUrl=""
  show={true}
>
  <styled.div
    show={true}
  >
    <div
      className="sc-bdnylx dRCoQz"
    >
      <p>
        Loading, please wait...
      </p>
      <img
        src="test-file-stub"
      />
    </div>
  </styled.div>
</LoadingOverlay>
`;
